tests/meta-mender-ci/recipes-testing/mender-mock-server/files/mender-mock-server.py

```diff
@@ -40,6 +41,21 @@ def do_GET(self):
                 self.end_headers()
 
                 shutil.copyfileobj(body, self.wfile)
+        elif re.match("/api/devices/v2/deployments/device/deployments/([a-f0-9-]*)/update_control_map$"):
+            if os.path.exists("/data/mender-mock-server-deployment-header.json"):
+                with open("/data/mender-mock-server-deployment-header.json", "rb") as body:
+                    # Extract only the update-control-map
+                    update = json.load(body.read())
+                    resp = json.dumps({
+                        "update_control_map": update["update_control_map"],
+                    }).encode("utf-8")
+                    self.send_response(200)
+                    self.send_header("Content-Type", "application/json")
+                    self.send_header("Content-Length", len(resp))
+                    self.end_headers()
+                    self.wfile.write(resp)
+            else:
+                self.send_empty_response(204)
```

Comment:
This endpoint [cannot return 204](https://github.com/mendersoftware/deployments-enterprise/blob/master/docs/devices_api_v2.yml#L127-L140). It should return 404 in that case (there is no map). In this case the client should not call it at all, but it's good to have it verified.